tests/unittest/test_util.py

```diff
@@ -121,56 +131,55 @@ def some_thing():
 
     def test_retry_retryable_exceptions(self):
         """Test retry retryable_exceptions arg does the right thing"""
-        sleeps = []
+        # This will fail on the first attempt and raise MyException because MyException
+        # is not in the list of retryable exceptions.
+        class MyException(Exception):
+            pass
 
-        def fake_sleep(attempt):
-            sleeps.append(attempt)
+        fake_sleep = make_fake_sleep()
 
-        # This will fail on the first attempt because Exception is not
-        # in the list of retryable exceptions.
-        @retry(retryable_exceptions=ValueError, sleep_function=fake_sleep)
+        @retry(retryable_exceptions=ValueError, sleep_function=make_fake_sleep)
         def some_thing():
-            raise Exception
+            raise MyException
 
-        with pytest.raises(Exception):
```

Comment:
`MaxAttemptsError` is a subclass of `Exception` so fixing this to throw and check for `MyException` removes the possibility that it's a `MaxAttemptsError` which should be a test failure.